core/types/transaction.go

```diff
@@ -440,6 +446,9 @@ func (tx *Transaction) WithoutBlobTxSidecar() *Transaction {
 	if !ok {
 		return tx
 	}
+	if sc := blobtx.Sidecar; sc != nil {
+		tx.sidecarSize = rlp.ListSize(sc.encodedSize())
+	}
```

Comment:
I haven't looked into this PR too deeply yet, but this part strikes me as really weird, and goes against the implied semantics of the method (which one would assume doesn't mutate the instance it is called on).

Reply:
I've fixed this. The intention is to set cpy.sidecarSize not the tx.sidecarSize.